sdk/lib/crt/math/arm/__stoi64.s

```diff
@@ -1,23 +1,36 @@
 /*
- * COPYRIGHT:         BSD - See COPYING.ARM in the top level directory
- * PROJECT:           ReactOS CRT library
- * PURPOSE:           Implementation of __stoi64
- * PROGRAMMER:        Timo Kreuzer (timo.kreuzer@reactos.org)
+ * PROJECT:     ReactOS CRT library
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     Implementation of __stoi64
+ * COPYRIGHT:   Copyright 2015 Timo Kreuzer <timo.kreuzer@reactos.org>
+ *              Copyright 2021 Roman Masanin <36927roma@gmail.com>
  */
 
 /* INCLUDES ******************************************************************/
 
 #include <kxarm.h>
 
+    IMPORT __stoi64_worker
+
 /* CODE **********************************************************************/
 
     TEXTAREA
 
+    /*
+        IN: s0 = single (float) value
+        OUT: r1:r0 = int64 value
+    */
     LEAF_ENTRY __stoi64
+    /* Allocate stack space and store parameters there */
+    push {lr}
+    PROLOG_END
 
-	__assertfail
-	bx	lr
+    /* Call the C worker function */
+    VMOV r0,s0
```

Comment:
Couldn't the same be done with passing a double to the worker function and using a union in there?

Reply:
I'm not sure, actually

Reply:
No, the union doesn't work in that way. But we can make something like using a pointer cast. But I don't think it's a good idea...

Reply:
Ok I added your idea :)